Quadriga/src/main/java/edu/asu/spring/quadriga/rest/PassThroughProjectRestController.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.domain.impl.passthroughproject.PassThroughProjectInfo;
+import edu.asu.spring.quadriga.domain.passthroughproject.IPassThroughProject;
+import edu.asu.spring.quadriga.exceptions.DocumentParserException;
+import edu.asu.spring.quadriga.exceptions.NoSuchRoleException;
+import edu.asu.spring.quadriga.exceptions.QStoreStorageException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.exceptions.RestException;
+import edu.asu.spring.quadriga.service.IRestMessage;
+import edu.asu.spring.quadriga.service.IUserManager;
+import edu.asu.spring.quadriga.service.network.INetworkManager;
+import edu.asu.spring.quadriga.service.passthroughproject.IPassThroughProjectDocumentReader;
+import edu.asu.spring.quadriga.service.passthroughproject.IPassThroughProjectManager;
+
+/**
+ * 
+ * Controller for pass through projects rest apis exposed to other clients
+ *
+ */
+@Controller
+public class PassThroughProjectRestController {
+
+    @Autowired
+    private IPassThroughProjectManager passThroughProjectManager;
+
+    @Autowired
+    private IUserManager userManager;
+
+    @Autowired
+    private IRestMessage errorMessageRest;
+
+    @Autowired
+    private IPassThroughProjectDocumentReader passThroughProjectDocumentReader;
+
+    @Autowired
+    private INetworkManager networkManager;
+
+    @RequestMapping(value = "rest/passthroughproject", method = RequestMethod.POST)
+    public ResponseEntity<String> getPassThroughProject(HttpServletRequest request,
+            @RequestHeader("Accept") String accept, HttpServletResponse response, @RequestBody String xml,
+            Principal principal) throws RestException {
+
+        xml = xml.trim();
+        if (xml.isEmpty()) {
+            String errorMsg = errorMessageRest.getErrorMsg("Please provide XML in body of the post request.");
+            return new ResponseEntity<String>(errorMsg, HttpStatus.BAD_REQUEST);
+
+        }
+        String userid = principal.getName();
+        String networkId = null;
+        try {
+            IUser user = userManager.getUser(userid);
+            PassThroughProjectInfo passThroughProjectInfo = passThroughProjectDocumentReader
+                    .getPassThroughProjectInfo(xml);
+
+            IPassThroughProject project = passThroughProjectDocumentReader
+                    .getPassThroughProject(passThroughProjectInfo);
+
+            String projectId = passThroughProjectManager.savePassThroughProject(user, project);
+
+            String workspaceId = passThroughProjectManager.createWorkspaceForExternalProject(passThroughProjectInfo,
+                    projectId, user);
+
+            String network = passThroughProjectDocumentReader.getNetwork(xml);
```

Comment:
It may be that there is no network xml submitted. This method shouldn't fail if that's the case, but simply create the project and return project id and workspace id. The client can then add networks to the created project in the next step.